src/vm/functions/conversions.rs

```diff
@@ -113,3 +114,93 @@ pub fn native_buff_to_uint_be(value: Value) -> Result<Value> {
     }
     return buff_to_int_generic(value, EndianDirection::BigEndian, convert_to_uint_be);
 }
+
+pub fn native_string_to_int_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Result<Value>,
+) -> Result<Value> {
+    match value {
+        Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data }))) => {
+            match String::from_utf8(data) {
+                Ok(as_string) => conversion_fn(as_string),
+                Err(_error) => Ok(Value::none()),
+            }
+        }
+        Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))) => {
+            let flattened_bytes = data.into_iter().flatten().collect();
+            match String::from_utf8(flattened_bytes) {
+                Ok(as_string) => conversion_fn(as_string),
+                Err(_error) => Ok(Value::none()),
+            }
+        }
+        _ => Err(CheckErrors::UnionTypeValueError(
+            vec![
+                TypeSignature::max_string_ascii(),
+                TypeSignature::max_string_utf8(),
+            ],
+            value,
+        )
+        .into()),
+    }
+}
+
+fn safe_convert_string_to_int(raw_string: String) -> Result<Value> {
+    let possible_int = raw_string.parse::<i128>();
+    match possible_int {
+        Ok(val) => return Value::some(Value::Int(val)),
+        Err(_error) => return Ok(Value::none()),
+    }
+}
+
+pub fn native_string_to_int(value: Value) -> Result<Value> {
+    native_string_to_int_generic(value, safe_convert_string_to_int)
+}
+
+fn safe_convert_string_to_uint(raw_string: String) -> Result<Value> {
+    let possible_int = raw_string.parse::<u128>();
+    match possible_int {
+        Ok(val) => return Value::some(Value::UInt(val)),
+        Err(_error) => return Ok(Value::none()),
+    }
+}
+
+pub fn native_string_to_uint(value: Value) -> Result<Value> {
+    native_string_to_int_generic(value, safe_convert_string_to_uint)
+}
+
+pub fn native_int_to_string_generic(
+    value: Value,
+    conversion_fn: fn(bytes: Vec<u8>) -> Result<Value>,
+) -> Result<Value> {
+    match value {
+        Value::Int(ref int_value) => {
+            let as_string = int_value.to_string();
+            match conversion_fn(as_string.into()) {
+                Ok(value) => Ok(value),
+                Err(_error) => Err(CheckErrors::InternalError.into()),
```

Comment:
Is it possible to reach this case? `int_value.to_string()` will always produce a string that would fit ASCII or UTF8, right? If it can't be reached, this should be switched to an `expects("PANIC: ...")` statement, and then we won't need the `InternalError` variant anymore.

Reply:
This call to "conversion_fn" should always work... that's why I was going to use InternalError.

It was based on the suggestion of @jcnelson not to use "unwrap" or "expect" in production code.

https://github.com/blockstack/stacks-blockchain/pull/2721#discussion_r656549977

Reply:
Hmmm, I think in this case, we want to expect or unwrap -- in general, we should expect/unwrap in places that should never fail: it's better for the process to end abruptly than continue to operate with broken assumptions. I think from the discussion linked, the main issue was for `string -> int`. But for `int -> string`, we should be confident that the conversion never fails.

Reply:
done